app/src/main/java/com/github/onlynotesswent/ui/user/PublicProfile.kt

```diff
@@ -191,8 +202,38 @@ private fun ProfileScaffold(
 fun TopProfileBar(
     title: String,
     navigationActions: NavigationActions,
+    userViewModel: UserViewModel,
     includeBackButton: Boolean = true,
-    onBackButtonClick: () -> Unit = { navigationActions.goBack() }
+    onBackButtonClick: () -> Unit = {
+      var userProfileId = navigationActions.popFromScreenNavigationStack()
+      // set profile user to userProfileId and navigate to public profile screen
+      // If we pop from stack and the profile id corresponds to profile user (we will navigate to
+      // the current screen),
+      // we pop twice to get to previous visited public profile
+      if (userProfileId != null && userProfileId == userViewModel.profileUser.value?.uid) {
+        userProfileId = navigationActions.popFromScreenNavigationStack()
+        if (userProfileId != null) {
```

Comment:
Why do you need to pop twice if the popped user is the same? I think this should not be needed if all calls to navactions are correct, bc no string would get pushed twice.

That being said, if you want to prevent going back to the same profile, consider this improvement on the code:

The nested if can be removed and the the large if-else flattened into two ifs, one that pops until userprofile is different than the current, and the second one that checks for null and navigates correctly. This reduces code duplication.

Alternatively, a new navigation actions method called .popUntilNew() or popWhileSame() could be created.  This is the better choice in my view

Reply:
The thing is I push to the stack when navigating to a user profile, so for the last user, I will push his id and then when I call pop I will get his id back. That's the reason for popping twice